tempesta_fw/t/functional/tests/test_frang.py

```diff
@@ -0,0 +1,516 @@
+#!/usr/bin/env python
+__author__ = 'Tempesta Technologies'
+__copyright__ = 'Copyright (C) 2016 Tempesta Technologies. (info@natsys-lab.com).'
+__license__ = 'GPL2'
+import sys
+from os.path import dirname, realpath, sep
+import time
+import types
+sys.path.append(dirname(realpath(__file__))+ sep + sep + "helpers")
+
+import conf
+import tfw
+from socket import *
+import select
+import struct
+
+global tcount
+
+tcount = 0
+
+class Test:
+	def __init__(self):
+		self.vs_get = b"GET / HTTP/1.0\r\nhost: loc\r\n\r\n"
+		self.s = socket(AF_INET, SOCK_STREAM)
+		self.cfg = conf.Config('etc/tempesta_fw.conf')
+		self.cfg.add_option('cache', '0')
+		self.cfg.add_option('listen', '8081')
+		self.cfg.add_option('server', '127.0.0.1:80')
+
+	def uri_len(self):
+		global tcount
+
+		self.res = False
+		print("uri\n")
+		self.__init__()
+		self.cfg.add_section('frang_limits')
+		self.cfg.add_option('http_uri_len', '3')
+		self.cfg.add_end_of_section()
+		self.vs_get = b"GET /index.html HTTP/1.0\r\nhost: loc\r\n\r\n"
+		tfw.start_with_frang()
+		self.s.connect(("127.0.0.1",8081))
+		print("tfw start\n")
+		try:
+			self.s.send(self.vs_get)
+			data = self.s.recv(1024)
+			if len(data) == 0:
+				self.res = True
+				tcount += 1
+		except OSError as e:
```

Comment:
Why the exception is not handled in many other similar places? It seems reading and verification of response should be moved to some generic function, such that the logic becomes more robust and consistent.